File: Java/src/demo/Node.java
package demo;

public class Node implements Comparable<Node>{
    int weight;
    byte data;
    Node left;
    Node right;

    public Node(Byte data, int weight) {
        this.data = data;
        this.weight = weight;
    }

    @Override
    public int compareTo(Node o) {
        return o.weight-this.weight;
    }

    @Override
    public String toString() {
        return "Node{" +
                "weight=" + weight +
                ", data=" + data +
                '}';
    }
}

File: Julia/src/Node.jl
"""
Java的手法,Julia的实现
"""
mutable struct Node
    left::Node
    right::Node
    weight::Int
    # byte
    data::UInt8

    function Node(data::UInt8, weight::Int)
        Node.data = data
        Node.weight = weight
    end

    function toString()
        return "Node{'
            weight = $weight
            , data = $data
            '}"
    end
end